cleanup-passes/v3-elements.ts

```diff
@@ -0,0 +1,172 @@
+ * be found at http://polymer.github.io/CONTRIBUTORS.txt Code distributed by
+ * Google as part of the polymer project is also subject to an additional IP
+ * rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import * as yaml from 'js-yaml';
+import {execFile} from 'child_process';
+import {promisify} from 'util';
+
+import {register} from '../cleanup-pass';
+import {ElementRepo} from '../element-repo';
+import {makeCommit} from './util';
+
+const execFilePromise = promisify(execFile);
+
+const filesWeMightChange = new Set<string>([
+  '.travis.yml',
+  '.npmignore',
+  '.gitignore',
+  'package.json',
+  'package-lock.json',
+]);
+
+/**
+ * Perform updates for 3.0 Polymer elements.
+ */
+async function v3ElementPass(element: ElementRepo): Promise<void> {
+  await updateNpmIgnore(element);
+  await updateGitIgnore(element);
+  await updateTravisYaml(element);
+  await updatePackageJson(element);
+  await regeneratePackageLock(element);
+
+  let shouldCommit = false;
+  for (const changedFile of await element.repo.getStatus()) {
+    const filePath = changedFile.path();
+    if (filesWeMightChange.has(filePath)) {
+      shouldCommit = true;
+    } else {
+      throw new Error(
+          `${element.ghRepo.name}: Unexpected changed file: ${filePath}`);
+    }
+  }
+  if (shouldCommit) {
+    await makeCommit(
+        element,
+        [...filesWeMightChange],
+        'Tedium automated v3 element updates.');
+  }
+}
+
+register({name: 'v3-elements', pass: v3ElementPass});
+
+const addToNpmIgnore = [
+  '*.tgz',
+  '.github',
+  '.travis.yml',
+  'formatconfig.json',
+  'gen-tsd.json',
+  'test/',
+  'wct.conf.json',
+];
+
+async function updateNpmIgnore(element: ElementRepo): Promise<void> {
+  const npmIgnorePath = path.join(element.dir, '.npmignore');
+  let npmIgnore = '';
+  if (await fse.pathExists(npmIgnorePath)) {
+    npmIgnore = await fse.readFile(npmIgnorePath, 'utf8');
+  }
+  for (const pattern of addToNpmIgnore) {
+    if (!npmIgnore.includes(pattern)) {
+      if (!npmIgnore.endsWith('\n') && npmIgnore !== '') {
+        npmIgnore += '\n';
+      }
+      npmIgnore += pattern + '\n'
+    }
+  }
+  await fse.writeFile(npmIgnorePath, npmIgnore);
+}
+
+const addToGitIgnore = [
+  '*.d.ts',
+  '*.tgz',
+  'node_modules',
+];
+
+async function updateGitIgnore(element: ElementRepo): Promise<void> {
+  const gitIgnorePath = path.join(element.dir, '.gitignore');
+  let gitIgnore = '';
+  if (await fse.pathExists(gitIgnorePath)) {
+    gitIgnore = await fse.readFile(gitIgnorePath, 'utf8');
+  }
+  for (const pattern of addToGitIgnore) {
+    if (!gitIgnore.includes(pattern)) {
+      if (!gitIgnore.endsWith('\n') && gitIgnore !== '') {
+        gitIgnore += '\n';
+      }
+      gitIgnore += pattern + '\n'
+    }
+  }
+  await fse.writeFile(gitIgnorePath, gitIgnore);
+}
```

Comment:
This function is almost duplicate apart from the filename and the dictionary. Maybe abstract on it?